src/pages/AnimeLib/main.ts

```diff
@@ -0,0 +1,261 @@
+    },
+  },
+  player: {
+    episode: 1,
+    total: 1,
+    season: 1,
+    next: undefined,
+  },
+};
+
+export const AnimeLib: pageInterface = {
+  domain: 'https://anilib.me',
+  languages: ['Russian'],
+  name: 'AnimeLib',
+  type: 'anime',
+  getImage() {
+    return anime.data.cover.default || anime.data.cover.thumbnail;
+  },
+  isSyncPage(url: string): boolean {
+    return utils.urlPart(url, 6) !== '' && utils.urlPart(url, 4) === 'anime';
+  },
+  isOverviewPage(url: string): boolean {
+    return utils.urlPart(url, 6) === '' && utils.urlPart(url, 4) === 'anime';
+  },
+  sync: {
+    getTitle(url: string): string {
+      return anime.data.eng_name || anime.data.name || anime.data.rus_name;
+    },
+    getIdentifier(url: string): string {
+      return anime.data.id.toString();
+    },
+    getOverviewUrl(url: string): string {
+      return utils.absoluteLink(`ru/anime/${anime.data.slug_url}`, AnimeLib.domain);
+    },
+    getEpisode(url: string): number {
+      return anime.player.episode;
+    },
+    nextEpUrl(url: string): string | undefined {
+      return anime.player.next;
+    },
+    getMalUrl(provider) {
+      if (anime.data.shikimori_href) {
+        const id = anime.data.shikimori_href.match(/\/[a-z]?(\d+)/);
+        if (id) {
+          return `https://myanimelist.net/anime/${id[1]}`;
+        }
+      }
+      return false;
+    },
+  },
+  overview: {
+    getTitle(url) {
+      return anime.data.eng_name || anime.data.name || anime.data.rus_name;
+    },
+    getIdentifier(url) {
+      return anime.data.id.toString();
+    },
+    uiSelector(selector) {
+      j.$('.tabs._border').before(j.html(selector));
+    },
+    getMalUrl(provider) {
+      return AnimeLib.sync.getMalUrl!(provider);
+    },
+  },
+
+  init(page: SyncPage) {
+    api.storage.addStyle(
+      require('!to-string-loader!css-loader!less-loader!./style.less').toString(),
+    );
+
+    j.$(() => {
+      utils.fullUrlChangeDetect(check, true);
+    });
+
+    async function check() {
+      page.reset();
+      clearInterval(interval);
+      if (
+        !AnimeLib.isSyncPage(window.location.href) &&
+        !AnimeLib.isOverviewPage!(window.location.href)
+      )
+        return;
+
+      // NOTE - We are on the SYNC page
+      if (AnimeLib.isSyncPage(window.location.href)) {
+        await updateSyncPage();
+
+        interval = utils.changeDetect(
+          async () => {
+            page.reset();
+            await updateSyncPage();
+            page.handlePage();
+          },
+          () => window.location.search,
+        );
+      }
+
+      // NOTE - We are on the OVERVIEW page
+      if (AnimeLib.isOverviewPage!(window.location.href)) {
+        await updateOverviewPage();
+
+        interval = utils.waitUntilTrue(
+          () => {
+            return j.$('.tabs-item').length;
+          },
+          () => {
+            page.handlePage();
+          },
+          500,
+        );
+      }
+
+      page.handlePage();
```

Comment:
This results in all overivew pages beeing triggered 2 times. I imagine you wanted inside of  https://github.com/MALSync/MALSync/pull/2606/files#diff-ad21d95b5e6745de2f7f26e52df7f79efe09ffce69150b485a72cb64e069ca45R102

Reply:
Ah, yes, this need to be in sync one